ansible_base/lib/middleware/logging/log_request.py

```diff
@@ -0,0 +1,23 @@
+import logging
+
+from ansible_base.lib.logging import thread_local
+
+logger = logging.getLogger(__name__)
+
+
+class LogRequestMiddleware:
+    """
+    Inject the request into the thread local so that it can be accessed by the logging filter.
+    """
+
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+    def __call__(self, request):
+        response = self.get_response(request)
+        thread_local.request = request
+        return response
+
+    def process_request(self, request):
+        thread_local.request = request
+        return None
```

Comment:
I think `process_request` is only called if subclassing `django.utils.deprecation.MiddlewareMixin`.

https://docs.djangoproject.com/en/5.0/topics/http/middleware/

If you're not, you could call `process_request` in the `__call__` method... or just write it like the modern examples.

Reply:
I only implemented this because we have code elsewhere that specifically calls `process_request()`, originally I didn't have this method. But yeah, could probably make `__call__` just use this method.